plugins/chat/spec/system/single_thread_spec.rb

```diff
@@ -266,5 +266,49 @@
         expect(page).to have_selector(".chat-thread .chat-message-separator__text", text: "Today")
       end
     end
+
+    context "when thread messages are sent with client timestamps" do
+      it "orders thread messages by client timestamp rather than server timestamp" do
+        first_message =
+          Chat::CreateMessage.call(
+            params: {
+              chat_channel_id: channel.id,
+              thread_id: thread.id,
+              message: "Thread message created first but should appear second",
+              client_created_at: 30.seconds.ago.iso8601,
+            },
+            guardian: current_user.guardian,
+          ).message_instance
+
+        second_message =
+          Chat::CreateMessage.call(
+            params: {
+              chat_channel_id: channel.id,
+              thread_id: thread.id,
+              message: "Thread message created second but should appear first",
+              client_created_at: 45.seconds.ago.iso8601,
+            },
+            guardian: current_user.guardian,
+          ).message_instance
+
+        chat_page.visit_thread(thread)
+
+        expect(page).to have_selector(
+          ".chat-thread .chat-message-container[data-id='#{first_message.id}'] .chat-message-text",
```

Comment:
You can use the `message_by_id` inside the `PageObjects::Pages::ChatThread` for this